src/test/java/io/confluent/idesidecar/restapi/kafkarest/RecordsV3ApiImplIT.java

```diff
@@ -276,6 +313,130 @@ void shouldThrowBadRequestForInvalidProtobufData(Object badData) {
           SchemaFormat.PROTOBUF, badData
       );
     }
+
+    static Stream<Arguments> unsupportedSchemaDetails() {
+      return Stream.of(
+          Arguments.of(
+              ProduceRequestData
+                  .builder()
+                  .data(Map.of())
+                  // Schema ID is not supported
+                  .schemaId(1)
+                  .build()
+          ),
+          Arguments.of(
+              ProduceRequestData
+                  .builder()
+                  .data(Map.of())
+                  // Passing raw schema is not supported
+                  .schema("invalid")
+                  // Passing schema type is not supported
+                  .type("AVRO")
+                  .build()
+          ),
+          Arguments.of(
+              ProduceRequestData
+                  .builder()
+                  .data(Map.of())
+                  // Passing schema version is supported
+                  .schemaVersion(1)
+                  // But type is not supported
+                  .type("PROTOBUF")
+                  .build()
+          ),
+          Arguments.of(
+              ProduceRequestData
+                  .builder()
+                  .data(Map.of())
+                  .schemaVersion(1)
+                  // Passing only subject is not supported
+                  .subject("standalone")
+                  .build()
+          ),
+          Arguments.of(
+              ProduceRequestData
+                  .builder()
+                  .data(Map.of())
+                  .schemaVersion(1)
+                  // Passing only subject name strategy is not supported
+                  .subjectNameStrategy("record_name")
+                  .build()
+          )
+      );
+    }
+
+    @ParameterizedTest
+    @MethodSource("unsupportedSchemaDetails")
+    void shouldThrowNotImplementedForUnsupportedSchemaDetails(ProduceRequestData data) {
+      var topic = randomTopicName();
+      createTopic(topic);
+      produceRecordThen(
+          topic,
+          ProduceRequest
+              .builder()
+              .partitionId(null)
+              // Doesn't matter if key or value, the schema details within
+              // should trigger the 501 response
+              .key(data)
+              .value(data)
+              .build()
+      )
+          .statusCode(501)
+          .body("message", equalTo(
+              "This endpoint does not support specifying schema ID, type, schema, standalone subject or subject name strategy."
+          ));
```

Comment:
Should this return a 4xx error because it's a client-side error?

Reply:
> Hmm, I think it's fine to leave it as "501 Not Implemented" since the client would expect to be able to pass these schema details. These are documented fields as part of the [kafka-rest produce records spec](https://github.com/confluentinc/kafka-rest/blob/master/api/v3/openapi.yaml#L2500-L2533). WDYT?

Hm, should we document these constraints in the OpenAPI spec?

Reply:
So, there's a few layers of API specifications to deal with here:

- [OpenAPI spec that the VS Code extension uses for Produce Records API](https://github.com/confluentinc/vscode/blob/main/src/clients/sidecar-openapi-specs/ce-kafka-rest.openapi.yaml#L979-L1034)
- Sidecar exposes wildcard route at `/kafka/v3/clusters*` -- this is not documented in the sidecar API spec.
- The wildcard route handler forwards the request to `/internal/kafka/v3/clusters/<cluster-id>/topics/<topic-name>/records` -- this is [explicitly excluded](https://github.com/confluentinc/ide-sidecar/blob/main/src/main/resources/application.yml#L245-L250) from the sidecar API spec.  

If we were to document these constraints, we'd have to manually update the OpenAPI spec the VS Code extension uses, which we actually have to do _anyway_ because it is currently modeled after the Confluent Cloud Kafka REST endpoint. The CCloud Kafka REST endpoint for Produce Records [does not accept schema details](https://github.com/confluentinc/vscode/blob/main/src/clients/sidecar-openapi-specs/ce-kafka-rest.openapi.yaml#L2624-L2655) (because it only supports TopicNameStrategy).

Reply:
Created https://github.com/confluentinc/vscode/issues/613 to track that work.

Reply:
Changed this to throw 400 instead of 501.